plutus-pab/src/Plutus/PAB/Webserver/Server.hs

```diff
@@ -112,8 +114,12 @@ startServer ::
     -> Either ClientEnv (PABAction t env WalletInfo)
     -> Availability
```

Comment:
@silky  Why not add a  `[Middleware]`  parameter here? Users can then call this function and set up their server accordingly. Same question for `startServerDebug`. These are the only two functions exposed by this module.

Reply:
`Middleware` is a `Monoid` so there is no need for a list

Reply:
@Jimbo4350 

The `Middleware` parameter is on `startServer'`; so I could expose that if it would help you?

I'm reluctant to put it on all the functions; because then all kinds of call-sites need to be updated; at least this way it's encapsulated nicely?

Reply:
>Middleware is a Monoid so there is no need for a list
Ah right I see the instance: ` Monoid b => Monoid (a -> b)`
>The Middleware parameter is on startServer'; so I could expose that if it would help you?
This would be more useful on the `startServerDebug` to be honest. Rather than update a bunch of call sites why not have separate functions that aren't tied to your cli? It would be nice to be able to customize the underlying warp server and have debugging available.

Reply:
@Jimbo4350 - The issue with putting it on `startServerDebug` is that the call-sites don't have a config file to read from; so I'd just have to default it so something by hand; which feels quite ugly.

I think one way to clean things up would be to have `Default` instance and use that a bit more; then it should be a bit clearer how all the functions interact? (i.e. `startServerDebug` just calls `startServer` with some default values.) ?

Reply:
>The issue with putting it on startServerDebug is that the call-sites don't have a config file to read from
That's fine. You could create `startServerDebugCustom` that leaves the configuration up to the caller.